KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,327 @@
+    kls_list = extraneous_section_deleter(kls_list)
+
+    kls_list.append("Limit Relation")
+    for item in kls_list:
+        if item not in kls_list_chap:
+            kls_list_chap.append(item)
+    kls_list_chap[kls_list_chap.index("Special value")] = "Symmetry"
+    kls_list_chap.append("Special value")
+    w = 2
+    h = len(kls_list_chap)+1
+    sorter_check = [[0 for _ in range(w)] for __ in range(h)]
+    return kls_list_chap, sorter_check
+
+
+def fix_chapter_sort(kls, chap, word, sortloc, klsaddparas, sortmatch_2, tempref, sorter_check):
+    """
+    This function sorts through the input files and identifies and places sections from the addendum after their
+    respective subsections in the chapter.
+
+    :param kls: The addendum where the sections to be inserted are found.
+    :param chap: The destination of inserted sections.
+    :param word: The keyword that is being processed.
+    :param sortloc: The location in the list of words, of the word being sorted.
+    :param klsaddparas:
+    :param sortmatch_2:
+    :return: This function outputs the processed chapter into a larger function for further processing.
+    """
+    hyper_headers_chap = []
+    hyper_subs_chap = []
+
+    special_input = 0
+    name_chap = word.lower()
+    print name_chap
+    if name_chap == "orthogonality":
+        special_input = 1
+    elif name_chap == "special value":
+        special_input = 2
+    elif name_chap == "symmetry":
+        special_input = 3
+    elif name_chap == "bilateral generating function":
+        special_input = 4
+
+    khyper_header_chap = []
+    k_hyper_sub_chap = []
+    index = 0
+
+    chapterstart = False
+    for item in kls:
+        index += 1
+        line = str(item)
+        if "\\subsection" in line:
+            temp = line[line.find(" ", 12)+1: line.find("}", 12)]  # get just the name (like mathpeople)
+            if temp.lower() == 'wilson':
+                chapterstart = True
+        if special_input in (0, 2, 4) and name_chap in line.lower() and chapterstart and ("\\paragraph{" in line or "\\subsubsection*{" in line) or \
+        (special_input in (1, 3) and "orthogonality relation" not in line.lower() and name_chap in line.lower() and chapterstart
+         and ("\\paragraph{" in line or "\\subsubsection*{" in line)):
+            for item in klsaddparas:
+                if index < item:
+                    klsloc = klsaddparas.index(item)
+                    break
+            t = ''.join(kls[index: klsaddparas[klsloc]])
+            k_hyper_sub_chap.append(t)  # append the whole paragraph, pray every paragraph ends with a % comment
+            khyper_header_chap.append(temp)  # append the name of subsection
+
+    for item in khyper_header_chap:
+        if item == "Pseudo Jacobi (or Routh-Romanovski)":
+            khyper_header_chap[khyper_header_chap.index(item)] = "Pseudo Jacobi"
+    k_hyp_index_iii = 0
+    offset = 0
+
+    item = 0
+    while item < len(khyper_header_chap):
```

Comment:
don't use item unless it is actually an item, and its a very generic function. Here i makes more sense because it's just an iterating variable, not an item in a list